### DISCUSSION TITLE
Select selected value in edit page.

### DISCUSSION THREAD

**Discussion by shankhadevpadam** [filamentphp/filament#3392] [Help]
I have two tables posts and postmeta. And I have a page template select field in the page resource. The issue is when saving the data the selected data in the page template select field is saved into the database. But when I edit the record, the selected page template is not selected in the page template select field.

```
posts table

|id           | title         | content 
| ----------  | -----------   | --------
| 1           | Title         | Content 
```

```
postmeta

|meta_id  | post_id | meta_key     | meta_value
|-------  | ------- | -------      | -------
|1        |1        |page_template |national
```

My issue is I want to select this meta value on the select field in the edit page.

![New Project](https://user-images.githubusercontent.com/3838722/183054868-84531d44-018e-4f6a-b092-6aa1c3bcaf50.jpg)

**Comment by Z3d0X** [comment]
Please provide the relevant code

**Reply by shankhadevpadam** [reply]
```php
Forms\Components\Select::make('page_template')
    ->label('Template')
    ->options([
        'default' => 'Default Template',
        'national' => "National",
    ])
    ->default('default')
    ->disablePlaceholderSelection()
```

**Reply by danharrin** [reply]
Do you have a cast for this property?

**Reply by shankhadevpadam** [reply]
No, this template are read from the template file like in wordpress.

**Reply by danharrin** [reply]
So the select should be saved and loaded from a relationship instead of the actual Post model?

You might need to wrap the Select in a Group with relationship() then

https://filamentphp.com/docs/2.x/forms/layout#saving-data-to-relationships

**Reply by shankhadevpadam** [reply]
But I have saved the data in key value chain. Please check my table structure.

**Reply by danharrin** [reply]
Well you need to load the data into the Eloquent model, otherwise Filament cant fill the form

IMO the meta table should be refactored into a relationship or add the columns to the posts table, its much easier

**Reply by shankhadevpadam** [reply]
Is there any method to select selected value in edit page.  Value may be query using custom query.

**Reply by shankhadevpadam** [reply]
I am just following the wordpress pattern for meta value.

**Reply by danharrin** [reply]
https://filamentphp.com/docs/2.x/admin/resources/editing-records#customizing-data-before-filling-the-form

you can access the post with `$this->record`

**Reply by danharrin** [reply]
> I am just following the wordpress pattern for meta value.

That is a bad pattern, so if you are just following it because Wordpress uses it, don't. Only use it if you're dealing with legacy data, and even then look to migrate to something better

**Reply by shankhadevpadam** [reply]
Can you make the selected method in the select class so the value passed in that method should be selected in the option dropdown.

**Reply by shankhadevpadam** [reply]
```php
Forms\Components\Select::make('page_template')
    ->label('Template')
    ->options([
        'default' => 'Default Template',
        'national' => "National",
    ])
    ->selected('national') // the selected parameter may be either integer, string, or closure
````

**Reply by chrispian** [reply]
I have the same issue. I have one table called comopany_roles and it has 4 fields (id, name, description, type). Type is just a text value saved as either Company or User. This is saving correctly. But when I edit, nothing in the dropdown is selected. Is there a way to mark one of the items selected based on its database value? I looked all through the Select source code and didn't see any logic for this but I may have missed it. Here is my code as well:

```php
/*
Below the 'type' is stored in the db field for this model (Role)
and I want to the select to pick the right one that's already been saved.
When a user edits, it's not picking either of them.
*/

public static function form(Form $form): Form
    {
        return $form
            ->schema([
                TextInput::make('name')
                         ->required(),
                         // ->unique('company_roles', 'name'),
                TextInput::make('description')
                         ->required(),
                Select::make('type')
                      ->options([
                          'User' => 'User',
                          'Company' => 'Company',
                      ])
                      ->required(),
            ])
            ->columns(1);
    }

```

**Reply by danharrin** [reply]
@chrispian you're doing everything right, so it sounds like a bug? Is there a cast on your model for that field?

The original question in this thread is simply an issue with bad database design, which I did explain.

**Reply by chrispian** [reply]
@danharrin Sorry, I didn't see the more replies option! Long day lol. There's no casts on it since it's just a text field, but I could be mistaken and it needs a specific cast?

**Reply by danharrin** [reply]
Nope, it should just work, the same as the other fields. Check the "status" select on the order resource in filamentphp/demo - it's the same.

So something is going wrong here. Capitalization of option keys? Do you have an accessor or anything? What happens if you swap the Select for TextInput, is the value loaded?

**Reply by danharrin** [reply]
I'm going to bed, if you cant work it out in a few hours (try debugging the $data using [this method](https://filamentphp.com/docs/2.x/admin/resources/editing-records#customizing-data-before-filling-the-form)), send a repo with the bug reproduction, and I will check it out in the morning.

**Reply by shankhadevpadam** [reply]
> ```
> Forms\Components\Select::make('page_template')
>     ->label('Template')
>     ->options([
>         'default' => 'Default Template',
>         'national' => "National",
>     ])
>     ->selected('national') // the selected parameter may be either integer, string, or closure
> ```

Is it possible to implement this concept?

**Reply by danharrin** [reply]
No. You cannot use the Wordpress meta table structure with Filament. Stick to Eloquent recommendations, with the template column on your pages table, instead of looking at Wordpress for inspiration.

Unfortunately, I don't have time to support you any longer with this issue, I've already spent quite a long time explaining why this is a bad idea.

**Reply by shankhadevpadam** [reply]
Ok, thanks @danharrin I will fix myself.

**Comment by chrispian** [comment]
@danharrin - Good catch! I switched it to a TextInput and can see that the data is full caps vs just the first character of the word - disregard - human error. TY!
